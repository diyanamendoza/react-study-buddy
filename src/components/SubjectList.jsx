import { Link } from 'react-router-dom'
import { data } from '../data/data'
import styles from '../App.css'
import { v4 as uuidv4 } from 'uuid'
import { motion } from 'framer-motion'
import { rSlideVariants } from '../variants'
import { useState } from 'react'

export default function SubjectList() {
  const [list, setList] = useState(filterResults ? filterResults : data)
  const [filterResults, setFilterResults] = useState('')
  const [filterInput, setFilterInput] = useState('')

  const filterItems = (query) => {
    console.log(query)
    setFilterInput(query.toLowerCase())
    const results = data.filter((item) => {
      const name = item.subject.toLowerCase()
      return name.includes(query)
    })
    console.log(results)
    setFilterResults(results)
    setList(results)
  }

  // const handleClear = () => {
  //   setFilterInput('')
  //   setFilterResults('')
  //   setList(data)
  // }

  return (
    <>
      <form>
        <label htmlFor="filter-items"></label>
        <input
          id="filter-items"
          type="text"
          name="filter-items"
          value={filterInput}
          autoComplete="off"
          onChange={({ target }) => filterItems(target.value)}
        />
        {/* <button
          type="button"
          id="btn"
          className="btn-filter"
          onClick={handleClear}
        >
          CLEAR
        </button> */}
      </form>
      <motion.nav
        variants={rSlideVariants}
        initial="initial"
        animate="animate"
        exit="exit"
      >
        {list.map((entry) => {
          let letterArr = Array.from(entry.subject)
          return (
            <Link to={`/${entry.subject}`} key={uuidv4()}>
              <div className={styles.subjcontainer}>
                {letterArr.map((letter) => {
                  return (
                    <motion.div
                      whileHover={{ scale: 1.5 }}
                      transition={{
                        type: 'tween',
                        ease: 'easeIn',
                        duration: 0.3,
                      }}
                      className={styles.letter}
                      key={uuidv4()}
                    >
                      {letter}
                    </motion.div>
                  )
                })}
              </div>
            </Link>
          )
        })}
      </motion.nav>
    </>
  )
}
