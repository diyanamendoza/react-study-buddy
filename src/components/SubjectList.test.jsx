import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router'
import SubjectList from './SubjectList'

it('should render the subject list', async () => {
  const { container } = render(
    <MemoryRouter>
      <SubjectList />
    </MemoryRouter>
  )

  expect(container).toMatchSnapshot()
})
