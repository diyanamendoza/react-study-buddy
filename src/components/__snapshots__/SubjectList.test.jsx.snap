// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the subject list 1`] = `
<div>
  <nav
    style="opacity: 0; transform: translateX(100vw) translateZ(0);"
  >
    <a
      href="/ECMAScript"
    >
      <div
        class="subjcontainer"
      >
        <div
          class="letter"
        >
          E
        </div>
        <div
          class="letter"
        >
          C
        </div>
        <div
          class="letter"
        >
          M
        </div>
        <div
          class="letter"
        >
          A
        </div>
        <div
          class="letter"
        >
          S
        </div>
        <div
          class="letter"
        >
          c
        </div>
        <div
          class="letter"
        >
          r
        </div>
        <div
          class="letter"
        >
          i
        </div>
        <div
          class="letter"
        >
          p
        </div>
        <div
          class="letter"
        >
          t
        </div>
      </div>
    </a>
    <a
      href="/Virtual DOM"
    >
      <div
        class="subjcontainer"
      >
        <div
          class="letter"
        >
          V
        </div>
        <div
          class="letter"
        >
          i
        </div>
        <div
          class="letter"
        >
          r
        </div>
        <div
          class="letter"
        >
          t
        </div>
        <div
          class="letter"
        >
          u
        </div>
        <div
          class="letter"
        >
          a
        </div>
        <div
          class="letter"
        >
          l
        </div>
        <div
          class="letter"
        >
           
        </div>
        <div
          class="letter"
        >
          D
        </div>
        <div
          class="letter"
        >
          O
        </div>
        <div
          class="letter"
        >
          M
        </div>
      </div>
    </a>
    <a
      href="/useReducer"
    >
      <div
        class="subjcontainer"
      >
        <div
          class="letter"
        >
          u
        </div>
        <div
          class="letter"
        >
          s
        </div>
        <div
          class="letter"
        >
          e
        </div>
        <div
          class="letter"
        >
          R
        </div>
        <div
          class="letter"
        >
          e
        </div>
        <div
          class="letter"
        >
          d
        </div>
        <div
          class="letter"
        >
          u
        </div>
        <div
          class="letter"
        >
          c
        </div>
        <div
          class="letter"
        >
          e
        </div>
        <div
          class="letter"
        >
          r
        </div>
      </div>
    </a>
    <a
      href="/useRef"
    >
      <div
        class="subjcontainer"
      >
        <div
          class="letter"
        >
          u
        </div>
        <div
          class="letter"
        >
          s
        </div>
        <div
          class="letter"
        >
          e
        </div>
        <div
          class="letter"
        >
          R
        </div>
        <div
          class="letter"
        >
          e
        </div>
        <div
          class="letter"
        >
          f
        </div>
      </div>
    </a>
    <a
      href="/useCallback"
    >
      <div
        class="subjcontainer"
      >
        <div
          class="letter"
        >
          u
        </div>
        <div
          class="letter"
        >
          s
        </div>
        <div
          class="letter"
        >
          e
        </div>
        <div
          class="letter"
        >
          C
        </div>
        <div
          class="letter"
        >
          a
        </div>
        <div
          class="letter"
        >
          l
        </div>
        <div
          class="letter"
        >
          l
        </div>
        <div
          class="letter"
        >
          b
        </div>
        <div
          class="letter"
        >
          a
        </div>
        <div
          class="letter"
        >
          c
        </div>
        <div
          class="letter"
        >
          k
        </div>
      </div>
    </a>
  </nav>
</div>
`;
